promql/parse.go

```diff
@@ -665,6 +660,9 @@ func (p *parser) primaryExpr() Expr {
 	case t.typ.isAggregator():
 		p.backup()
 		return p.aggrExpr()
+
+	default:
+		p.errorf("no valid expression found")
 	}
 	return nil
```

Comment:
You could remove this line now since it's dead code (legal in Go if there's a default in the switch).